include/ui.js

```diff
@@ -171,18 +171,6 @@ var UI;
                 UI.updateViewDrag();
                 UI.setBarPosition();
             } );
-
-            // While connected we want to display a confirmation dialogue
-            // if the user tries to leave the page
-            Util.addEvent(window, 'beforeunload', function (e) {
-                if (UI.rfb && UI.rfb_state === 'normal') {
```

Comment:
See here: https://developer.mozilla.org/en/docs/Web/Events/beforeunload
In short. Firefox doesn't show the message. Some mobile browsers doesn't show it all. The HTML standard seems to be moving towards not allowing any blocking of leaving a page.